plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardFragment.java

```diff
@@ -356,22 +366,81 @@ public void widgetSelected(SelectionEvent e) {
 		};
 	}
 
+	private String substitueVariables(String rawSourcePath) {
+		if (org.apache.commons.lang.StringUtils.isNotBlank(rawSourcePath)) {
+			try {
+				return VariablesPlugin.getDefault().getStringVariableManager().performStringSubstitution(rawSourcePath);
+			} catch (CoreException swalloed) {
+				OpenShiftUIActivator.getDefault().getLogger().logError("Could not interpolate "+rawSourcePath);
+			}
+		}
+		return rawSourcePath;
+	}
+
 	private SelectionAdapter onBrowseWorkspace(final Shell shell) {
 		return new SelectionAdapter() {
 
 			@Override
 			public void widgetSelected(SelectionEvent e) {
-				DirectoryDialog dialog = new DirectoryDialog(shell, SWT.OPEN);
-				dialog.setText("Choose the source path to sync");
-				dialog.setFilterPath(ResourcesPlugin.getWorkspace().getRoot().getLocation().toString());
-				String filepath = dialog.open();
-				if (!StringUtils.isEmpty(filepath)) {
-					model.setSourcePath(filepath);
+				String sourcePath = getWorkspaceRelativePath(model.getSourcePath());
+				ElementTreeSelectionDialog dialog = createWorkspaceFolderDialog(shell, sourcePath);
+				if (dialog.open() == IDialogConstants.OK_ID && dialog.getFirstResult() instanceof IContainer) {
+					String path = ((IContainer)dialog.getFirstResult()).getFullPath().toPortableString();
+					String folderPath = VariablesPlugin.getDefault().getStringVariableManager().generateVariableExpression("workspace_loc", path);
+					model.setSourcePath(folderPath);
 				}
 			}
 		};
 	}
 
+	private String getWorkspaceRelativePath(String sourcePath) {
+		if (org.apache.commons.lang.StringUtils.isBlank(sourcePath) || sourcePath.contains("{")) {
+			return sourcePath;
+		}
+		IPath absolutePath = new Path(sourcePath);
+		IContainer container = ResourcesPlugin.getWorkspace().getRoot().getContainerForLocation(absolutePath);
+		if (container != null) {
+			return container.getFullPath().toPortableString();
```

Comment:
I am not sure that toPortableString() is preferred here to just toString().
IPath.toString() makes only separator '/' for all platforms and it is what we need.
Docs say that if toPortableString() is used it has to be matched with fromPortableString() when converting back to Path.